specs/governance/gov-token.md

```diff
@@ -24,72 +24,82 @@
 
 `GovernanceToken` is an [ERC20](https://eips.ethereum.org/EIPS/eip-20) token contract that inherits from `ERC20Burnable`,
 `ERC20Votes`, and `Ownable`. It allows token holders to delegate their voting power to other addresses, enabling a representative
-voting system.
+voting system. The contract integrates with the `Alligator` contract through a hook-based approach to enable advanced delegation
+functionality, including partial delegation and subdelegations.
 
-The specific implementation details of the voting and delegation logic are inherited from the `ERC20Votes` contract. The
-`GovernanceToken` contract focuses on integrating this functionality with `ERC20Burnable` and adding the minting
-capability. EIP-712 permit functionality is also inherited from `ERC20Permit`, which is imported by `ERC20Votes`.
+### Hook-based Integration with Alligator
 
-### Checkpoints
+The `GovernanceToken` contract integrates with the `Alligator` contract through a hook-based approach to enable advanced
+delegation features. The `_afterTokenTransfer` function in the `GovernanceToken` is modified to call the `afterTokenTransfer`
+function in the `Alligator` contract, allowing the `Alligator` to consume the hooks and update its delegation and checkpoint
+mappings accordingly.
 
-`ERC20Votes` uses checkpoints to track voting power history. A checkpoint is a "snapshot" of the voting power of an
-address at a specific block number.
+To ensure that token transfers can continue even if the call to the Alligator fails, a try-catch mechanism is implemented
+in the `_afterTokenTransfer` function. If the call to the Alligator's `afterTokenTransfer` function fails, the token transfer
+will still be completed, and an event will be emitted to indicate the failure. This prevents the token transfer from being
+blocked due to issues with the Alligator contract.
 
 ```solidity
-struct Checkpoint {
-    uint32 fromBlock;
-    uint224 votes;
+function _afterTokenTransfer(address from, address to, uint256 amount) internal override {
+    bytes memory data = abi.encodeWithSelector(IAlligator.afterTokenTransfer.selector, from, to, amount);
+    bool success;
+    assembly ("memory-safe") {
+        success := call(gas(), alligatorAddress, 0, add(data, 0x20), mload(data), 0, 0)
+    }
+    
+    if (!success) emit AlligatorCallFailed(from, to, amount);
 }
 ```
```

Comment:
In the same spirit of not leaking implementation details I would delete this Solidity block, I don't think it adds value and it makes the spec harder to maintain. I would instead suggest adding a sentence defining the event signature

---

To expand a bit so I don't think we've discussed this much, the rationale for "not leaking implementation details into the spec" is:
- Simplify spec maintenance: By specifying as little as possible to implement the requirements, we make it easier to keep the spec up to date by having less code to maintain. The code in this spec is not tested or audited, and we may end up tweaking it.
- Readability: The less code, the easier it is for a less-technical reader to understand the spec and ensure it's sound.
- Testing: The implementer of the spec has flexibility around the final result, as long as it meets the requirements. When implementing the spec, we write tests for all the requirements in here, but we wouldn't write a test to e.g. ensure we're using assembly since that's an implementation detail and not a requirement